File: array and string/largest_number_at_least_twice_of_others.java
class Solution {
    public int dominantIndex(int[] nums) {
        int big = nums[0];
        int index=0;
        
        for(int i=0; i < nums.length; i++)
        {
            if (nums[i]>big)
            {
                big = nums[i];
                index = i;
            }
        }
        
        for(int i=0; i < nums.length; i++)
        {
            if (big < 2*nums[i] && nums[i]!=big)
            {
                return -1;
            }
        }
        
        return index;
    }
}

File: array and string/largest_number_at_least_twice_of_others.py
class Solution:
    def dominantIndex(self, nums: List[int]) -> int:
        big = max(nums)
        ind = nums.index(max(nums))
        for i in range(len(nums)):
            if big < 2*nums[i] and big!=nums[i]:
                return -1
        return ind